platforms/software/dependency-management/src/main/java/org/gradle/internal/component/local/model/DefaultLocalComponentMetadata.java

```diff
@@ -327,6 +329,11 @@ public Set<String> getConfigurationNames() {
 
         @Override
         public void visitConfigurations(Consumer<Candidate> visitor) {
+            GradleException failure = VariantIdentityUniquenessVerifier.buildReport(configurationsProvider).aggregateFailure();
```

Comment:
💭 You could switch this to return `Optional<GradleException>` and the use a more functional style chaining `.ifPresent()` to avoid returning null.

Reply:
In general I am not convinced that using `Optional` is any better than using `null`. Since we use null much more frequently throughout the codebase, I'd prefer to continue using that -- at least we should decide as a team to transition to using `Optional` more generally. 

In any case, I've updated the method to throw an exception instead of returning a value, so this doesn't matter here anymore